### PR TITLE
feat: separate useControlledState hook

### PR DESCRIPTION
**Pull request by IsaevAlexandr**
@GermanVor related to https://github.com/gravity-ui/uikit/pull/1090#discussion_r1459544901.

I'm not sure if this functionality is needed or not. I make this proposal to discuss

### CODE REVIEW

**Comment by GermanVor** [comment]
also [here](https://github.com/gravity-ui/uikit/pull/1090#discussion_r1459507200)

I think it's a simple hook that will be regularly used while developing library components

File: src/hooks/useControlledState/index.ts [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import React from 'react';
+
+export function useControlledState<T>(value: T | undefined, defaultValue: T) {
+    const [state, setState] = React.useState(value ?? defaultValue);
+
+    return [
+        value || state,
```
Comment (GermanVor) [inline src/hooks/useControlledState/index.ts:7]: `value ?? state`

**Comment by GermanVor** [comment]
@amje @korvin89 Ping

**Comment by korvin89** [comment]
@IsaevAlexandr Hey! We already have [this one](https://github.com/gravity-ui/date-components/blob/main/src/components/hooks/useControlledState.ts). Could we use your hook instead of date-components hook?

**Comment by amje** [comment]
@IsaevAlexandr @korvin89 Seems the same functionality is implemented in #1276

**Comment by korvin89** [comment]
This functionality have added here https://github.com/gravity-ui/uikit/pull/1276

**Pull request closed without merging**